authz/sdk_server_interceptors.go

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package authz
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/internal/xds/rbac"
+	"google.golang.org/grpc/status"
+)
+
+// StaticInterceptor contains engines used to make authorization decisions. It
+// either contains two engines deny engine followed by an allow engine or only
+// one allow engine.
+type StaticInterceptor struct {
+	engines rbac.ChainEngine
+}
+
+// NewStatic returns a new StaticInterceptor from a static authorization policy
+// JSON string.
+func NewStatic(authzPolicy string) (*StaticInterceptor, error) {
+	RBACPolicies, err := translatePolicy(authzPolicy)
+	if err != nil {
+		return nil, err
+	}
+	chainEngine, err := rbac.NewChainEngine(RBACPolicies)
+	if err != nil {
+		return nil, err
+	}
+	if chainEngine.IsEmpty() {
+		return nil, fmt.Errorf("failed to initialize RBAC engines")
+	}
```

Comment:
Why do you need this?  Shouldn't `err != nil` be returned if it could not be initialized?

Reply:
That's correct. Currently its a redundant sanity check. If the translator executes successfully, then we would have at-least one rbac policy, thereby creating at-least one engine. I had added the check, thinking about future changes, where we could end up creating an interceptor with no engine (and would prefer returning an error then). Removed the checks since it is redundant now.